tsdb/store_test.go

```diff
@@ -144,6 +144,54 @@ func TestStore_CreateShard(t *testing.T) {
 	}
 }
 
+// Ensure the store can create a new shard.
+func TestStore_StartupShardProgress(t *testing.T) {
+	t.Parallel()
+
+	test := func(index string) {
+		fmt.Println(index)
+		s := MustOpenStore(index)
+		defer s.Close()
+
+		// Create a new shard and verify that it exists.
+		require.NoError(t, s.CreateShard("db0", "rp0", 1, true))
+		sh := s.Shard(1)
+		require.NotNil(t, sh)
+
+		// Create another shard and verify that it exists.
+		require.NoError(t, s.CreateShard("db0", "rp0", 2, true))
+		sh = s.Shard(2)
+		require.NotNil(t, sh)
+
+		msl := &mockStartupLogger{}
+
+		// Reopen shard and recheck.
+		require.NoError(t, s.ReopenWithStartupMetrics(msl))
+		sh = s.Shard(1)
+		require.NotNil(t, sh)
+
+		// Create another shard and verify that it exists.
+		require.NoError(t, s.CreateShard("db0", "rp0", 2, true))
+		sh = s.Shard(2)
+		require.NotNil(t, sh)
+
+		// Equality check to make sure shards are always added prior to
+		// completion being called.
+		require.Equal(t, msl.shardTracker, []string{
+			"shard-add",
+			"shard-add",
+			"shard-add",
+			"shard-complete",
+			"shard-complete",
+			"shard-complete",
+		})
```

Comment:
At the time `ReopenWithStartupMetrics` is called there are only 2 shards in the store. Why are there 3 calls each to `shard-add` and `shard-complete` instead of just 2 calls each?

Reply:
There are three shards actually just one is in an error state. It will still be counted when attempting to be opened though. Wanted to display that. Might be good for me to write a comment.

Reply:
There are only 2 shards. Did you check what directory the third "shard" is in? 

While working on main-2.x branch I discovered there's been a long standing issue with the test scaffolding: https://github.com/influxdata/influxdb/blob/96bade409e73709bd01f9cadf1d745767887bfeb/tsdb/store_test.go#L2630-L2634

That being said, I really like the idea of doing another Reopen with an invalid shard sitting out there. It improves the code coverage and makes sure you get to 100% even with bad shards. It just needs to be an intentional test and not one due to a problem with the scaffolding that I'm sure will be fixed at some point.

Reply:
I'll take a look - I thought that was what the test I copied this from was doing. Will take a look.